src/gateways/KeplrWalletGateway.ts

```diff
@@ -0,0 +1,28 @@
+import { ConnectionError } from 'domain/wallet/entities/errors'
+import { Wallet, WalletId } from 'domain/wallet/ports/walletPort'
+import { Account, Accounts, ChainId } from 'domain/wallet/entities/wallet'
+import { KeplrAccountDTO } from './DTO/keplrDTO'
+import { KeplrAccountMapper } from './mappers/account.mapper'
+
+export class KeplrWallet implements Wallet {
+  public isConnected = (): boolean =>
+    !!window.Keplr.getOfflineSigner && !!window.Keplr
+
+  public connect = (chainId: ChainId): Promise<void | ConnectionError> => {
+    if (!this.isConnected) {
+      return Promise.resolve(new ConnectionError())
+    }
+    return window.Keplr.enable(chainId)
```

Comment:
Tested in the console, the `enable()` can throw exceptions (I don't think it's clearly described in the official documentatin). So better wrap with a try/catch construct.

```
> window.keplr.enable("foo")
Promise {<pending>}
injectedScript.bundle.js:10 Uncaught (in promise) Error: There is no chain info for foo
    at r (injectedScript.bundle.js:10:32583)
```